apps/bfd-server/bfd-server-war/src/test/java/gov/cms/bfd/server/war/r4/providers/BeneficiaryTransformerV2Test.java

```diff
@@ -66,225 +77,296 @@ public void setup() {
+  }
+
+  /**
+   * Verifies that {@link gov.cms.bfd.server.war.r4.providers.BeneficiaryTransformerV2} correctly
+   * handles patient deceased attributes.
+   */
+  @Test
+  public void shouldMatchBeneficiaryDeathDate() {
+    DateTimeType deceasedDate = patient.getDeceasedDateTimeType();
+    if (deceasedDate != null) {
+      Assert.assertEquals("1981-03-17", deceasedDate.getValueAsString());
     } else {
-      Assert.assertEquals(1, patient.getAddress().size());
-      Assert.assertEquals(beneficiary.getStateCode(), patient.getAddress().get(0).getState());
-      // assert CountyCode is no longer mapped
-      Assert.assertNull(patient.getAddress().get(0).getDistrict());
-      Assert.assertEquals(beneficiary.getPostalCode(), patient.getAddress().get(0).getPostalCode());
-      // assert address city name and line 0 - 5 fields etc.
-      Assert.assertNull(patient.getAddress().get(0).getCity());
-      Assert.assertEquals(0, patient.getAddress().get(0).getLine().size());
+      Assert.assertEquals(new BooleanType(false), patient.getDeceasedBooleanType());
     }
   }
 
   /**
-   * Verifies that the {@link Patient} identifiers contain expected values.
-   *
-   * @param Patient {@link Patient} containing identifiers
-   * @param identifierSystem value to be matched
-   * @param identifierValue value to be matched
+   * Verifies that {@link gov.cms.bfd.server.war.r4.providers.BeneficiaryTransformerV2} works
+   * correctly when passed a {@link Beneficiary} without includeAddressFields header values.
    */
-  private static void assertValuesInPatientIdentifiers(
-      Patient patient, String identifierSystem, String identifierValue) {
-    boolean identifierFound = false;
-
-    for (Identifier temp : patient.getIdentifier()) {
-      if (identifierSystem.equals(temp.getSystem()) && identifierValue.equals(temp.getValue())) {
-        identifierFound = true;
-        break;
-      }
-    }
-    Assert.assertEquals(
-        "Identifier "
-            + identifierSystem
-            + " value = "
-            + identifierValue
-            + " does not match an expected value.",
-        identifierFound,
-        true);
+  @Test
+  public void shouldMatchAddressNoAddrHeader() {
+    List<Address> addrList = patient.getAddress();
+    Assert.assertEquals(1, addrList.size());
+    Address compare = new Address();
+    compare.setPostalCode("12345");
+    compare.setState("MO");
+    Assert.assertTrue(compare.equalsDeep(addrList.get(0)));
   }
 
   /**
-   * @return the {@link StaticRifResourceGroup#SAMPLE_A} {@link Beneficiary} record, with the {@link
-   *     Beneficiary#getBeneficiaryHistories()} and {@link
-   *     Beneficiary#getMedicareBeneficiaryIdHistories()} fields populated.
+   * Verifies that {@link gov.cms.bfd.server.war.r4.providers.BeneficiaryTransformerV2} works
+   * correctly when passed a {@link Beneficiary} with includeAddressFields header values.
    */
+  @Test
+  public void shouldMatchAddressWithAddrHeader() {
```

Comment:
More just for my knowledge: what's going on with needing a matching with address header in addition to matching without an address header? Looks like the serialized sample A output only has state and postal code, so I was interested to see this extra test.

Reply:
So the default will be to only return state and postal; however, if the RequestHeader has include address set to true, we return the entire address data.  So there is one test for the default, and one test for the entire address info.